lib/sync/startup.ts

```diff
@@ -42,10 +48,36 @@ export const syncRepoStartupListener: StartupListener = async ctx => {
     if (!cluster.isMaster) {
         return;
     }
-    await repoSync(sdm);
+    await sdmRepoSync(sdm);
     const interval: number = _.get(sdm, "configuration.sdm.k8s.options.sync.intervalMinutes");
     if (interval && interval > 0) {
-        setInterval(() => repoSync(sdm), interval * 60 * 1000);
+        sdm.addTriggeredListener({
+            trigger: { interval: interval * 60 * 1000 },
+            listener: async li => sdmRepoSync(li.sdm),
+        });
     }
     return;
 };
+
+/**
+ * Create the trappings required for executing a command and then
+ * create a job to execute the [[kubernetesSync]] command.
+ */
+async function sdmRepoSync(sdm: SoftwareDeliveryMachine): Promise<void> {
+    const disposers: Array<() => Promise<void>> = [];
+    const workspaceId = sdm.configuration.workspaceIds[0];
+    const context = fakeContext(workspaceId);
+    context.graphClient = sdm.configuration.graphql.client.factory.create(workspaceId, sdm.configuration);
+    context.lifecycle = {
+        dispose: async () => { await Promise.all(disposers.map(d => d())); },
+        registerDisposable: (d: () => Promise<void>) => { disposers.push(d); },
+    };
+    logger.info(`Creating sync repo job`);
+    await createJob({ command: kubernetesSync, parameters: [] }, context);
```

Comment:
createJob will schedule a job in the backend that comes back into the SDM via an event subscription. The execution of the wrapped command will be completely separate to this code here. You can remove all that disposable stuff. That will happen automatically.